test/entity/entity_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entity
+
+import (
+	"fmt"
+	"log"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/util/awsservice"
+)
+
+const (
+	sleepForFlush = 180 * time.Second
+
+	entityType        = "@entity.KeyAttributes.Type"
+	entityName        = "@entity.KeyAttributes.Name"
+	entityEnvironment = "@entity.KeyAttributes.Environment"
+
+	entityPlatform          = "@entity.Attributes.PlatformType"
+	entityInstanceId        = "@entity.Attributes.EC2.InstanceId"
+	entityEKSCluster        = "@entity.Attributes.EKS.Cluster"
+	entityK8sNode           = "@entity.Attributes.K8s.Node"
+	entityK8sNamespace      = "@entity.Attributes.K8s.Namespace"
+	entityK8sWorkload       = "@entity.Attributes.K8s.Workload"
+	entityServiceNameSource = "@entity.Attributes.AWS.ServiceNameSource"
+)
+
+type expectedEntity struct {
```

Comment:
can't this just be the `cloudwatch.Entity` struct? Then we can have checks to ensure the KeyAttributes/Attributes are valid in the struct?

Reply:
When querying the logs, the response doesn't return in `cloudwatch.Entity` type, it's `Results [][]*ResultField` which is basically arrays of strings. We could try to use `cloudwatch.Entity` but it would require additional logics to deference the maps just to turn them into strings so thought using a custom test struct is cleaner.